projects/plugins/boost/app/class-jetpack-boost.php

```diff
@@ -121,6 +121,32 @@ public function deactivate() {
 		do_action( 'jetpack_boost_deactivate' );
 		Regenerate_Admin_Notice::dismiss();
 		Analytics::record_user_event( 'deactivate_plugin' );
+		$this->deactivation_survey();
+	}
+
+	/**
+	 * Plugin deactivation survey handler. The plugin will have been deactivated but present this
+	 * page to ask for feedback.
+	 */
+	public function deactivation_survey() {
+		// Skip the deactivation feedback if it's a JSON/AJAX request or via WP-CLI
+		if ( wp_is_json_request() || wp_doing_ajax() || ( defined( 'WP_CLI' ) && WP_CLI ) || wp_is_xml_request() ) {
+			return;
+		}
+
+		if ( ! defined( 'BOOSTDEACTIVATINGINPROG' ) ) {
+			define( 'BOOSTDEACTIVATINGINPROG', true );
```

Comment:
Is this gating to prevent `deactivate_plugins()` from triggering this block of code again, creating a potentially infinite loop?

A few thoughts: 

1) I'd prefer to use local static variables over defining a const, as they are scoped to the function so it's impossible for there to ever be a scope collision,
2) This will only gate the call to deactivate_plugins. `jetpack_boost_deactivate` may get fired multiple times, we may record multiple analytics events, etc.
3) I'm a little uncomfortable at hijacking the deactivation event, then manually deactivating the plugin.